docling_eval/evaluators/ocr_evaluator.py

```diff
@@ -0,0 +1,153 @@
+        ds = load_dataset("parquet", data_files={split: split_files})
+        _log.info("Overview of dataset: %s", ds)
+
+        # Select the split
+        ds_selection: Dataset = ds[split]
+
+        text_evaluations_list = []
+        char_accuracy_list = []
+
+        for i, data in tqdm(
+            enumerate(ds_selection),
+            desc="Evaluating OCR",
+            ncols=120,
+            total=len(ds_selection),
+        ):
+            data_record = DatasetRecordWithPrediction.model_validate(data)
+            doc_id = data_record.doc_id
+            if data_record.status not in self._accepted_status:
+                _log.error(
+                    "Skipping record without successfull conversion status: %s", doc_id
+                )
+                continue
+
+            true_doc = data_record.ground_truth_doc
+            pred_doc = data_record.predicted_doc
+
+            if not pred_doc:
+                _log.error("There is no prediction for doc_id=%s", doc_id)
+                continue
+
+            true_text = self._extract_text(true_doc)
+            pred_text = self._extract_text(pred_doc)
+
+            if true_text and pred_text:
+                cer = self._compute_cer_score(true_text, pred_text)
+                char_accuracy = 1.0 - cer
+            else:
+                cer = 1.0  # max error when text is missing
+                char_accuracy = 0.0  # zero accuracy when text is missing
+
+            char_accuracy_list.append(char_accuracy)
+
+            page_evaluation = PageOcrEvaluation(
+                doc_id=doc_id,
+                true_text=true_text,
+                pred_text=pred_text,
+                cer=cer,
+                char_accuracy=char_accuracy,
+            )
+
+            text_evaluations_list.append(page_evaluation)
+            if self._intermediate_evaluations_path:
+                self.save_intermediate_evaluations(
+                    evaluation_name="ocr_eval",
+                    enunumerate_id=i,
+                    doc_id=doc_id,
+                    evaluations=[page_evaluation],
+                )
+
+        mean_character_accuracy = (
+            statistics.mean(char_accuracy_list) if char_accuracy_list else 0.0
+        )
+
+        _log.info(f"Mean Character Accuracy: {mean_character_accuracy:.4f}")
+
+        return DatasetOcrEvaluation(
+            evaluations=text_evaluations_list,
+            mean_character_accuracy=mean_character_accuracy,
+        )
+
+    def _compute_cer_score(self, true_txt: str, pred_txt: str) -> float:
+        """Compute Character Error Rate"""
+        result = self._cer_eval.compute(predictions=[pred_txt], references=[true_txt])
+        return result
+
+    def _extract_text(self, doc: DoclingDocument) -> str:
+        """Extract text from document JSON structure"""
+        extracted_text = ""
+        if hasattr(doc, "texts") and doc.texts:
+            for text_item in doc.texts:
+                if hasattr(text_item, "text"):
```

Comment:
`hasattr` should have no place in typesafe code. Hence, lines 149 and 151 should be simply removed.

Reply:
Addressed it myself.